### PR TITLE
✨ Add ERC1967I in LibClone

### PR DESCRIPTION
**Pull request by atarpara**
## Description

closing #821.

Describe the changes made in your pull request here.

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge fmt`?
- [x] Ran `forge snapshot`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

**Comment by atarpara** [comment]
@Vectorized can you check once bytecode bcs i am worried about `msize` trick for returning address. It can be compatible `zkSync`?

File: test/LibClone.t.sol [hunk 1]
```diff
@@ -77,10 +77,34 @@ contract LibCloneTest is SoladyTest, Clone {
...
         testDeployERC1967(1);
     }
 
+    function testDeployERC1967I() public {
+        testDeployERC1967I(1);
+    }
+
+    function testDeployERC1967ISpecialPath(bytes1 data) public {
+        address clone = LibClone.deployERC1967I(address(this));
+        (, bytes memory rd) = clone.call(abi.encodePacked(data));
```
Comment (wilsoncusack) [inline test/LibClone.t.sol:98]: confused, should this only work if data = "I"?

Comment (atarpara) [reply test/LibClone.t.sol:98]: It will return implementation as long `calldatasize` is `1`.

Comment (atarpara) [reply test/LibClone.t.sol:98]: It will return implementation as long `calldatasize` is `1`.

Comment (wilsoncusack) [reply test/LibClone.t.sol:98]: Thanks @atarpara , can you link me to this check in the code

**Comment by Vectorized** [comment]
Ok, this looks fantastic.

**Comment by Vectorized** [comment]
@atarpara Thanks for spotting the comment typos too.

**Pull request merged**